test/e2e/cloud-provider-azure.go

```diff
@@ -66,11 +66,15 @@ func InstallCalicoAndCloudProviderAzureHelmChart(ctx context.Context, input clus
 }
 
 // InstallAzureDiskCSIDriverHelmChart installs the official azure-disk CSI driver helm chart
-func InstallAzureDiskCSIDriverHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput) {
+func InstallAzureDiskCSIDriverHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, hasWindows bool) {
 	specName := "azuredisk-csi-drivers-install"
-	By("Installing the correct version of azure-disk CSI driver components via helm")
+	By("Installing azure-disk CSI driver components via helm")
 	options := &helmVals.Options{
-		Values: []string{"windows.useHostProcessContainers=true", "controller.replicas=1", "controller.runOnControlPlane=true"},
+		Values: []string{"controller.replicas=1", "controller.runOnControlPlane=true"},
+	}
+	// TODO: make this always true once HostProcessContainers are on for all supported k8s versions.
+	if hasWindows {
```

Comment:
Should we propose moving this logic into the official helm chart as well? (before injecting host process stuff into the actual resource spec you must enable the `windows.useHostProcessContainers` chart config *and* be installing onto a valid version of Kubernetes)

Reply:
That's what the Helm chart is validating actually (as part of the daemonset) and why that option wasn't valid and failing the Helm chart install for templates that didn't enable to correct feature flag for k8s < 1.26

Reply:
Oh got it, thanks.

Reply:
Do we support anything less than 1.22 now?

Reply:
No and in fact I disabled this for anything < 1.23 (https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2915/files#diff-2c7fc9c9c12632c5e550abeb31614356abd17710d948a8664f9d2f2163e38b0dR266). This is to cover the case between 1.23 and 1.26 where HostProcess containers still require the feature flag  for this to work (otherwise the Helm install fails), so this makes sure we don't break e2e templates that don't have windows (and thus might not have the feature flag set, which is the case for some of the CSI migration test templates).